### PR TITLE
Support GHC 9.4

### PR DESCRIPTION
**Pull request by parsonsmatt**

### CODE REVIEW

File: Data/Memory/Internal/CompatPrim64.hs [hunk 1]
```diff
@@ -63,6 +63,7 @@ type OutBool = Bool
 type Word64# = Word#
 type Int64# = Int#
 
+#if __GLASGOW_HASKELL__ < 904
```
Comment (parsonsmatt) [inline Data/Memory/Internal/CompatPrim64.hs:66]: Relevant note in migration guide: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.4#word64int64-use-word64int64

**Comment by benz0li** [comment]
GHC 9.4.1 has been released yesterday: https://www.haskell.org/ghc/download_ghc_9_4_1.html

@vincenthz What about GHC 9.4 support for hs-memory?

**Comment by mpilgrem** [comment]
I have sort of tested this pull request by using the fork to build Stack with GHC 9.4.1 (Stack depends on `memory`).

**Comment by ysangkok** [comment]
@parsonsmatt  The `cabal.project` file can now be removed from this PR. I just tested that it builds and passes without it.

File: Data/Memory/Internal/CompatPrim64.hs (hunk 1 above)
Comment (lf-) [reply Data/Memory/Internal/CompatPrim64.hs:66]: this conditional seems rather suspicious for a 9.4 compat change. Is it correct?

Comment (tfausak) [reply Data/Memory/Internal/CompatPrim64.hs:66]: The `904` here means 9.4, not 9.0.4 as you might expect. See the documentation for a more thorough explanation: https://downloads.haskell.org/~ghc/9.4.2/docs/users_guide/phases.html#standard-cpp-macros

**Comment by vincenthz** [comment]
thanks @parsonsmatt

**Comment by parsonsmatt** [comment]
Thank you @vincenthz 😄

**Pull request merged**